packages/core/src/browser/shell/application-shell.ts

```diff
@@ -344,6 +414,17 @@ export class ApplicationShell extends Widget {
         );
     }
 
+    /**
+     * Change the state of the application to currently loading (`true`) or ready (`false`).
+     * This has an impact on the behavior of certain operations, e.g. animations are disabled
+     * while loading.
+     */
+    set isLoading(value: boolean) {
```

Comment:
fyi: boolean property names usually don't have ´is` prefix, just `loading`

Reply:
Is this API suppose to be for all clients?

Reply:
It should be used _only_ in FrontendApplication. If you have a better idea for notifying the shell about the end of the startup phase, feel free to share it :-)